core/staking_verifier.go

```diff
@@ -98,10 +101,9 @@ func VerifyAndEditValidatorFromMsg(
 		return nil, errCommissionRateChangeTooHigh
 	}
 
-	// TODO: make sure we are reading from the correct snapshot
 	snapshotValidator, err := chainContext.ReadValidatorSnapshot(wrapper.Address)
```

Comment:
so we don't need to ensure its at right snapshot epoch?

Reply:
It's already ensured by the snapshot code being correct. The todo was added when the feature was first implemented.